File: app/src/main/java/ys/moire/MoireApplication.java
package ys.moire;

import android.app.Application;

import lombok.Getter;
import ys.moire.common.di.ApplicationComponent;
import ys.moire.common.di.ApplicationModule;
import ys.moire.common.di.DaggerApplicationComponent;

/**
 * Application class.
 */

public class MoireApplication extends Application {

    ApplicationComponent component;

    @Override
    public void onCreate() {
        super.onCreate();

        component = DaggerApplicationComponent
                .builder()
                .applicationModule(new ApplicationModule(this))
                .build();
    }

    public ApplicationComponent getComponent() {
        return component;
    }

}

File: app/src/main/java/ys/moire/MoireApplication.kt
package ys.moire

import androidx.multidex.MultiDexApplication
import com.google.android.gms.ads.MobileAds
import ys.moire.common.di.ApplicationComponent
import ys.moire.common.di.ApplicationModule
import ys.moire.common.di.DaggerApplicationComponent

/**
 * Application class.
 */

class MoireApplication : MultiDexApplication() {

    companion object {
        @JvmStatic lateinit var component: ApplicationComponent
    }

    override fun onCreate() {
        super.onCreate()

        component = DaggerApplicationComponent
                .builder()
                .applicationModule(ApplicationModule(this))
                .build()

        MobileAds.initialize(this)
    }

}
